src/OpenIddict.Core/OpenIddictExtensions.cs

```diff
@@ -49,9 +49,10 @@ public static class OpenIddictExtensions {
             [NotNull] this OpenIddictBuilder builder,
             [NotNull] string name, int position,
             [NotNull] Action<IApplicationBuilder> registration) {
-            // By default, prevent duplicate registrations.
-            if (builder.Modules.Any(module => string.Equals(module.Name, name))) {
-                return builder;
+            // By default, replace duplicate registrations.
+            var replacedModule = builder.Modules.Where(module => string.Equals(module.Name, name)).ToList();
```

Comment:
Actually you DO need a `ToList()`. If not, the foreach freaks out saying that the collection was modified and it can't iterate. Linq queries don't actually enumerate until the last second. Inside a foreach doesn't always work right. The `ToList()` forces it to enumerate before we use the foreach on the resultant collection, ensuring that the results can't change while the foreach is iterating. I tried without it :)